Scripts/RemediationScripts/Remediate-SetServiceBusMinReqTLSVersion.ps1

```diff
@@ -0,0 +1,779 @@
+
+        $validServiceBusNampespaces = $resourceDetails.FailedResourceList | Where-Object { ![String]::IsNullOrWhiteSpace($_.ResourceId) }	
+        if (($resourceDetails | Measure-Object).Count -eq 0 -or ($validServiceBusNampespaces | Measure-Object).Count -eq 0) {
+
+            Write-Host "No Azure Service Bus Namespace(s) found in input json file for remediation." -ForegroundColor $([Constants]::MessageType.Error)	
+            Write-Host $([Constants]::DoubleDashLine)	
+            return	
+        }
+        $validServiceBusNampespaces | ForEach-Object { 	
+            try {
+                $currentServiceBusNamespaceDetails = Get-AzServiceBusNamespace -ResourceGroupName $_.ResourceGroupName -NamespaceName $_.ResourceName -ErrorAction SilentlyContinue
+                $currentTLSVersion = ($currentServiceBusNamespaceDetails).MinimumTlsVersion
+                $serviceBusNamespacesDetails += $currentServiceBusNamespaceDetails  | Select-Object @{N = 'ResourceId'; E = { $_.Id } },
+                @{N = 'ResourceGroupName'; E = { $_.Id.Split("/")[4] } },
+                @{N = 'ResourceName'; E = { $_.Name } }, 
+                @{N = 'MinimumTLSVersion'; E = { $currentTLSVersion } }
+            }
+            catch {
+                Write-Host "Valid resource id(s) not found in input json file. Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Warning)	
+                Write-Host "Skipping the Resource:  [$($_.ResourceName)]..." -ForegroundColor $([Constants]::MessageType.Warning)	
+                $logResource = @{}
+                $logResource.Add("ResourceGroupName", ($_.ResourceGroupName))	
+                $logResource.Add("ResourceName", ($_.ResourceName))	
+                $logResource.Add("Reason", "Valid resource id(s) not found in input json file.")    	
+                $logSkippedResources += $logResource	
+                Write-Host $([Constants]::SingleDashLine)
+                Write-Host "Error fetching Azure Service Bus Namespace for resource: Resource ID:  [$($resourceId)]. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+            }	
+        }	
+    }
+    else {	
+        if ([String]::IsNullOrWhiteSpace($InputFilePath)) {
+            Write-Host "Fetching all Azure Service Bus Namespace(s) in Subscription: [$($context.Subscription.SubscriptionId)]" -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+
+            # Get all Azure Serivce Bus Namespace(s) in a Subscription
+            $serviceBusNamespaces = @();
+            $serviceBusNamespaces = Get-AzServiceBusNamespace -ErrorAction Stop
+            $serviceBusNamespaces | ForEach-Object { 	
+                $currentTLSVersion = $_.MinimumTlsVersion 
+                $serviceBusNamespacesDetails += $_  | Select-Object @{N = 'ResourceId'; E = { $_.Id } },
+                @{N = 'ResourceGroupName'; E = { $_.Id.Split("/")[4] } },
```

Comment:
Kindly shift these three lines to the right for readability purposes.

Reply:
Corrected.